### ISSUE TITLE
What a shame this is still broken

### ISSUE THREAD

**Issue by seamusdemora** [lazarus-recovery/lazarus_addon#24]
I've come here to learn what happened to Lazarus for form recovery. The dates mentioned in the README file leave me with little hope this will ever be recovered. That's too bad - it was apparently a great tool - and badly needed... I just lost two hours of work on a Stack Exchange piece I was working on in Firefox.

**Comment by NormG1** [comment]
Lazarus is dead for sure. Try Form History Control as an add on. I use Firefox and it works there. Not nearly as elegant as Lazarus, but it is better than nothing. Both Lazurus and Form History Control were BROWSER add ons. If you are working on say e-mail or WordPress, neither of these will help you. They only work in BROWSERs. Good luck.

**Comment by maphew** [comment]
Another option is to use an external editor which has built in autosave/undo. It's a partial solution since it doesn't do all form data but can work well for situations like the above. 

* https://github.com/GhostText/GhostText - multiplatform but must have specific configurations work with specific editors
* https://www.listary.com/text-editor-anywhere - Windows only, freeware.

Note: I haven't personally used these. I used to rely heavily on the "It's All Text" spiritual predecessor though, now [sadly discontinued](https://github.com/docwhat/itsalltext).

Another out of band solution is to use a long-term clipboard manager like **[CopyQ](https://hluk.github.io/CopyQ/)** and develop the habit of reflexively slapping `Ctrl-A`, `Ctrl-C` to "save", somewhat like we used to have to do with almost everything. Isn't _progress_ wonderful?

**Comment by NormG1** [comment]
> I've come here to learn what happened to Lazarus for form recovery. The dates mentioned in the README file leave me with little hope this will ever be recovered. That's too bad - it was apparently a great tool - and badly needed... I just lost two hours of work on a Stack Exchange piece I was working on in Firefox.

Yes, it is a shame, because there is a real need for something like this. I'm sorry you lost your work. It is maddening.

I hate spending 30 minutes crafting a reader comment for a New York Times article, pressing submit, and discovering they closed all comments in the background and I just wasted my time. 

I wrote to the NY Times to ask that closing comments means a user cannot START a new comment. But if there is a comment in progress, the user should be allowed to finish it and submit it with a time limit.  The user would receive an announcement. "New comments have been closed. Please complete your comment within the next 15 minutes and submit. After 15 minutes, you will no longer be able to submit your comment." But trying to talk tech win their customer support is a useless exercise. These people deal with subscriptions and not much else. 

I am old enough to remember time-outs. We used to pay for bandwidth by the minute, so forgetting to log out could be expensive. Now, supposedly, time outs are for security. But back in the day, getting timed out without warning meant you lost your work. So programmers started adding warnings . . . "You are going to be disconnected in x minutes. To stay connected, click here"

Well, if you are a programmer, maybe you could write a new program that does what Lazarus did. All the previous code is probably usable since multiple attempts were made to contact the original developer of Lazarus. I think he must have died because he cannot be found anywhere.

You develop it, you own it, you could make it a commercial product and charge for it but you'd need to continue to maintain it. Go for it!

**Comment by seamusdemora** [comment]
@NormG1:  

> Yes, it is a shame, because there is a real need for something like this. I'm sorry you lost your work. It is maddening.

I'm having good luck with another tool/addon that has similar objectives: [Form History Control (II)](https://stephanmahieu.github.io/fhc-home/) by Stephan Mahieu. You may also find it useful :)

**Comment by herreidler** [comment]
Guys, I have been followed this article including the attempting to reactive the addon. In all these dialogs, I remember that some versions were released inside this article. I kept one of them as xpi. I have been used it and works as well. Sure, you need to install it manually as a .xpi file. If someone is interested in this file, please let me know and I can create a link to get it. 
Best regards! Firefox forever!

**Comment by seamusdemora** [comment]
@herreidler :  The easiest thing to do would be to create a new GitHub repo, and put it up there.

**Comment by herreidler** [comment]
@seamusdemora Nice! Sorry for being a noob in Github. How can I do that? If you rather, I can send the xpi file via email too.

**Comment by seamusdemora** [comment]
@herreidler : Setting up a GitHub account is easy enough... I think [this page covers most of what you need to know.](https://docs.github.com/en/free-pro-team@latest/github/getting-started-with-github/quickstart). I'm probably not the best one to take it on as I have zero experience with FireFox add-ons, and I have no spare time at this point. If you don't wish to take this on, the [README file](https://github.com/lazarus-recovery/lazarus_addon) lists a point-of-contact. But I will say this again: the [Form History Control (II)](https://stephanmahieu.github.io/fhc-home/) works pretty well IMHO.

**Comment by Janaue** [comment]
interested in a new version for quantum

**Comment by flawrist** [comment]
@herreidler Is your xpi version is newer than [2017-Dec-18](https://github.com/lazarus-recovery/lazarus_addon/releases/tag/v3.2.2.692)?
